src/Microsoft.Extensions.Configuration.AzureAppConfiguration/FeatureManagement/FeatureManagementKeyValueAdapter.cs

```diff
@@ -44,9 +44,17 @@ public FeatureManagementKeyValueAdapter(FeatureFilterTracing featureFilterTracin
                 //if (featureFlag.Conditions?.ClientFilters == null)
                 if (featureFlag.Conditions?.ClientFilters == null || !featureFlag.Conditions.ClientFilters.Any()) // workaround since we are not yet setting client filters to null
                 {
-                    //
-                    // Always on
-                    keyValues.Add(new KeyValuePair<string, string>(featureFlagPath, true.ToString()));
+                    // Add the AlwaysOn filter instead of setting the flag to "true" so feature management doesn't skip evaluating variants and telemetry when present
+                    if (featureFlag.Variants != null || featureFlag.Telemetry != null)
+                    {
+                        keyValues.Add(new KeyValuePair<string, string>($"{featureFlagPath}:{FeatureManagementConstants.EnabledFor}:{0}:{FeatureManagementConstants.Name}", FeatureManagementConstants.AlwaysOnFilter));
+                    }
+                    else
+                    {
+                        //
+                        // Always on
```

Comment:
Could this comment be updated to call out "featureFlagPath: true" since "Always on" sounds like the filter, which is now handled differently.